staynomad/Nomad-Back
# Nodemailer Email Styling

# Relevant issue

Closes #15 

# Summary of change


* I accidentally branched from `friend-requests` so please refer to that too (#142 )
* Also let me know if I missed any emails
* Updated formatting for following emails: 
   * Listing Expiration
   * Account Verification
   * Listing Creation Confirmation
   * Transfer Invite
   * Transfer Acceptance (2 emails: one to sending and one to receiving party)
   * Transfer Rejection (2 emails: one to sending and one to receiving party)
   * Reservation Confirmation for Guest
   * Reservation Confirmation for Host
   * Checkin for Guest
   * Checkin for Host
   * Checkout for Guest
   * Checkout for Host
* Everything is a function for easy access. Pass `HTMLOptions` to `getHTML` to get the formatted HTML, and call `getAttachments` to get the attachments for the email. 
* Updated husky precommit hook for backend to not call `git add .` anymore
* Rebased with master

# Testing/Verification

* navigate to the `emailtester.js` file in `/helpers` directory. 
* change the name and email to your own name and email
* run `node emailtester.js` and check your inbox 
* emails should be the same format as the template, in `template.html`. 
     * links should be white on the gray background with no highlighting
     * you should get 12 emails total 
* please also test on mobile and other email clients
* check that i didnt break anything else please
